typify-impl/src/type_entry.rs

```diff
@@ -146,7 +146,11 @@ pub(crate) enum TypeEntryDetails {
     Option(TypeId),
     Box(TypeId),
     Vec(TypeId),
-    Map(TypeId, TypeId),
+    Map {
+        map_to_use: String,
```

Comment:
Does this need to be something we carry around for each TypeEntryDetails::Map? Could we instead resolve it in `type_ident`? That is to say, it seems like we're copying the value from `TypeSettings` in every map when we might be able to use the value from `TypeSettings` when rendering the type.

Reply:
Yes, we can. I originally did it this way to allow extension for individual endpoints to set the map type via `x-rust-type`. 

Will switch to a fully global method.